docs/cookbooks/deploying_aws.md

```diff
@@ -0,0 +1,590 @@
+      WORKING_DIR: "./"
+    steps: 
+    
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: linux/amd64
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@master
+        with:
+          platforms: linux/amd64
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build & Push ECR Image
+        uses: docker/build-push-action@v2
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          context: ${{ env.WORKING_DIR }}
+          platforms: linux/amd64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          tags: ${{ env.AWS_ECR_REPOSITORY }}:latest
+
+      - name: Deploy to ECS
+        run: |
+          aws ecs update-service --cluster ${{ env.AWS_ECS_CLUSTER_NAME }} --service ${{ env.AWS_ECS_SERVICE_NAME }} --desired-count 1 --force-new-deployment
+        env:
+          AWS_DEFAULT_REGION: ${{ env.AWS_REGION }}
+
+
+```
+
+# Step 4: Using with SDK
+
+You should be able to get the URL for your Guardrails API using:
+
+```bash
+export GUARDRAILS_BASE_URL=$(terraform output -raw backend_service_url)
+echo "http://$GUARDRAILS_BASE_URL"
+```
+
+By setting the above environment variable `GUARDRAILS_BASE_URL` the SDK will be able to use this as a backend for running validations.
+
+
+# Considerations for Production
+
+When building the container we made some crucial choices in order to make Guardrails production ready. Namely we opted for remote inferencing & did not include local models. This allows us to reduce the image size by preventing heavy validators from being included in the final container image and instead use hosted versions of ML based validators.
+
+In addition to the configuration above we recommend that you host your ECS cluster in private subnet and configure autoscaling policies with appropriate minimum & maximum container counts based on your expected workloads. Furthermore, we recommend that you also serve the API using https.
+
+# Terraform
+
+```hcl
+locals {
+    deployment_name = "gr-backend"   
+}
+
+variable "aws_region" {
+  description = "AWS region to deploy the resources"
+  type        = string
+  default     = "us-east-2"
+}
+
+variable "backend_cpu" {
+  description = "CPU units for the service"
+  type        = number
+  default     = 8*1024
+}
+
+variable "backend_memory" {
+  description = "Memory units for the service"
+  type        = number
+  default     = 16*1024
+}
```

Comment:
If we're only installing two validators these should be able to be scaled _way_ down.

Reply:
yeah good catch, I wasn't using remote inference option before

Reply:
set to 1 vcpu, and 2GB mem and using 3 threads for gunicorn